### DISCUSSION TITLE
Can't dump Bonk's Revenge for TurboGrafx-16

### DISCUSSION THREAD

**Discussion by dirn** [sanni/cartreader#798] [Need help]
I haven't been able to dump Bonk's Revenge for tg16. I've tried two different HuCards. I've tried using HuCARD(swapped), HuCARD(not swapped), and Turbochip. I've also tried forcing 1024kb for the two HuCard options.

I'm using the adapter from Save the Hero Builders and I have the switches set to the SNES settings.

I've successfully dumped other TurboGrafx and PC Engine games, but I can't get this one to work. Has anyone had any luck?

P.S. Using the HuCARD(not swapped) and 1024kb, I'm able to see the title and mode select screens, as well as the Start Round 1 screen, but after that it turns to artifacts.

<img width="988" alt="image" src="https://github.com/sanni/cartreader/assets/168109/c7587cb4-e2fe-434e-959b-90d1aa122ec1">

**Comment by PsyK0p4T** [comment]
Hey ! Just to be 100% sure, the carts you tried to dump are both this one right ?
![image](https://github.com/sanni/cartreader/assets/87064902/2b77c35a-5f39-42b5-9089-619f5e64aed6)
If they are, then the option to select in menu is "Turbochip" indeed.

- What hardware and software version are you using ?
- What is the rom size you get without forcing it to 1024k ? Expected size should be 512k according to Datomatic entries.
- After multiple redumps, is the CRC32 of your rom always the same ? And also the same between your two different carts ?

Don't forget to submit your redumps to No-Intro. Mostly the tg16 ones, many of them are not confirmed yet ;)

**Comment by dirn** [comment]
Yep, that's the same card.

I sat down to answer your questions and wanted to verify the rom sizes and that the checksums were in fact the same for both cards, and my first run dumped cleanly. Oddly it came through as the Virtual Console version but it definitely works.

I've never submitted anything to No-Intro. I'll look into that.

Thanks for taking the time to write out those questions. I'll keep them in mind if I run into trouble with anything else.

**Reply by PsyK0p4T** [reply]
Ok, let me recap things... I just want to be sure to get you well, because it could change things in the databases.

- you redumped once again both carts you have, and managed to finally get a "known" CRC. Which value was it ?
- Did you end up with that CRC more than once ? with the two carts ? If not, it would be very cool if you could do some more redumps. By cleaning the carts pins the best you can (like with a soft rubber) or also playing a bit with the cart position inside the slot, to have perfect contact. The goal here would be to regulary get that crc, with the two cartridges.

As you can see [here](https://datomatic.no-intro.org/index.php?page=show_record&s=12&n=0429) , the current state of Bonk's Revenge tg16 version is still "Not verified", with a lot of mess in the various entries... hacked or modified data etc. It would be great to finally get a clean/confirmed dump report for it. Mostly if you own twice the game physically.

Thanks for you time and reports :)
And to submit things to no-intro, just post your results and pictures in [that forum](https://forum.no-intro.org/viewforum.php?f=7)

**Reply by dirn** [reply]
> Ok, let me recap things... I just want to be sure to get you well, because it could change things in the databases.
> 
> * you redumped once again both carts you have, and managed to finally get a "known" CRC. Which value was it ?
> * Did you end up with that CRC more than once ? with the two carts ? If not, it would be very cool if you could do some more redumps. By cleaning the carts pins the best you can (like with a soft rubber) or also playing a bit with the cart position inside the slot, to have perfect contact. The goal here would be to regulary get that crc, with the two cartridges.

I dumped both carts, each one multiple times. The checksum was consist each time I dumped each cart, and the two carts dumped with different checksums.

The known one dumped with 

    f36ffee1
    34e8b8c01d7721a33c7d131f77c8c082
    a9908331d835ce3f6ae301e2a6232fe59861c454

It came through as `Bonk's Revenge (USA, Europe) (Virtual Console).pce`.

The unknown one dumped with

    68782a2f
    bbe5e014f8e2ed849bc844088d9c4c4b
    89d2540df8e761d2c2387cc082ca8c826e143491

I haven't played the game to completion, but it starts and is playable. I'll probably play around with this one some more to see if cleaning it better changes anything.

> 
> As you can see [here](https://datomatic.no-intro.org/index.php?page=show_record&s=12&n=0429) , the current state of Bonk's Revenge tg16 version is still "Not verified", with a lot of mess in the various entries... hacked or modified data etc. It would be great to finally get a clean/confirmed dump report for it. Mostly if you own twice the game physically.
> 
> Thanks for you time and reports :) And to submit things to no-intro, just post your results and pictures in [that forum](https://forum.no-intro.org/viewforum.php?f=7)

My forum access was approved overnight so I'm going to try to get my information posted later today.

**Reply by PsyK0p4T** [reply]
Thanks a lot for all explanations :)

Getting the CRC "f36ffee1" is a very nice info. It would proove that all previous entries dumped from a physical cart were bad in db.

Do you have accesss to Discord ? I would be interested to speak with you and look at your second dump, with crc "68782a2f" to compare it with others. It could be a new revision (low chances if it doesn't boot up in an emulator) or a bad dump (it's happening time to time, some cartridges are suffering from their age and some sectors start to be corrupted).

**Reply by dirn** [reply]
> Thanks a lot for all explanations :)
> 
> Getting the CRC "f36ffee1" is a very nice info. It would proove that all previous entries dumped from a physical cart were bad in db.
> 
> Do you have accesss to Discord ?

I'm dirn#5856.

> I would be interested to speak with you and look at your second dump, with crc "68782a2f" to compare it with others. It could be a new revision (low chances if it doesn't boot up in an emulator)

It does boot up and I can play it, though I didn't try to complete a level, I was just seeing if it worked at all.

> or a bad dump (it's happening time to time, some cartridges are suffering from their age and some sectors start to be corrupted).

I ran a quick script over the two files and compared them in chucks of 1024 bytes. Of the 513 chunks, only 43 didn't match.
